src/ParallelAlgorithms/SurfaceFinder/SurfaceFinder.cpp

```diff
@@ -0,0 +1,78 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "ParallelAlgorithms/SurfaceFinder/SurfaceFinder.hpp"
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "DataStructures/ApplyMatrices.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Index.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "Domain/Domain.hpp"
```

Comment:
You don't need this (anymore). Please delete this include and also remove `Domain` from the linked libraries in the CMakeLists.txt.